pkg/deploy/api/validation/validation.go

```diff
@@ -503,6 +503,18 @@ func IsValidPercent(percent string) bool {
 
 const isNegativeErrorMsg string = `must be non-negative`
 
+func ValidateDeploymentRequest(req *deployapi.DeploymentRequest) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(req.Name) == 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("name"), req.Name, "name of the deployment config is missing"))
+	} else if len(kvalidation.IsDNS1123Subdomain(req.Name)) != 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("name"), req.Name, "name of the deployment config is invalid"))
+	}
+
+	return allErrs
```

Comment:
Regarding admission the intermediate solution is going to be admission on
the nested client or moving the fields the separate subsets.

On Fri, Jul 22, 2016 at 5:11 PM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/deploy/api/validation/validation.go
> <https://github.com/openshift/origin/pull/9349#discussion_r71946812>:
>
> > @@ -503,6 +503,18 @@ func IsValidPercent(percent string) bool {
> >
> >  const isNegativeErrorMsg string = `must be non-negative`
> >
> > +func ValidateDeploymentRequest(req *deployapi.DeploymentRequest) field.ErrorList {
> > +	allErrs := field.ErrorList{}
> > +
> > +	if len(req.Name) == 0 {
> > +		allErrs = append(allErrs, field.Invalid(field.NewPath("name"), req.Name, "name of the deployment config is missing"))
> > +	} else if len(kvalidation.IsDNS1123Subdomain(req.Name)) != 0 {
> > +		allErrs = append(allErrs, field.Invalid(field.NewPath("name"), req.Name, "name of the deployment config is invalid"))
> > +	}
> > +
> > +	return allErrs
>
> The validation registration expects the validation function to have just
> the objecct tha tis validated as an argument.
>
> [mkargaki@dhcp129-211 origin]$ make test-unit WHAT=pkg/api/validation/
> TEST_KUBE=true GOTEST_FLAGS="" hack/test-go.sh pkg/api/validation/
> panic: expected one 'in' param, got: func(*api.DeploymentRequest, *api.DeploymentConfig) field.ErrorList
>
> goroutine 1 [running]:
> panic(0x182d560, 0xc4202903a0)
>     /usr/lib/golang/src/runtime/panic.go:500 +0x1aegithub.com/openshift/origin/pkg/api/validation.(*RuntimeObjectsValidator).MustRegister(0x25c1d50, 0x2595980, 0xc42007d280, 0x1808140, 0x1b4ef10, 0x0, 0x0)
>     github.com/openshift/origin/pkg/api/validation/_test/_obj_test/validation.go:44 +0x10dgithub.com/openshift/origin/pkg/api/validation.registerAll()
>     github.com/openshift/origin/pkg/api/validation/_test/_obj_test/register.go:71 +0x1021github.com/openshift/origin/pkg/api/validation.init.1()
>     github.com/openshift/origin/pkg/api/validation/_test/_obj_test/register.go:42 +0x48github.com/openshift/origin/pkg/api/validation.init()
>     /home/mkargaki/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/pkg/api/validation/validation_test.go:261 +0x7aa
> main.init()
>     github.com/openshift/origin/pkg/api/validation/_test/_testmain.go:118 +0xad
> FAIL    github.com/openshift/origin/pkg/api/validation  0.071s
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9349/files/ac3fd615856d9d0c50eea0208191cbd0122cb319..af47c3c1f6b70da9f98ca5a69db57106b126e6c2#r71946812>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_pzst2jCyPmd_GAVyCXbOLae3nW7kks5qYTIWgaJpZM4I2lSD>
> .
>